api/indexer/client.go

```diff
@@ -27,6 +29,54 @@ type Client struct {
 	requester *requester.EndpointRequester
 }
 
+// Use a separate type that only decodes the block bytes because we cannot decode block JSON
+// due to Actions/Auth interfaces included in the block's transactions.
+type GetBlockClientResponse struct {
+	BlockBytes codec.Bytes `json:"blockBytes"`
```

Comment:
Yup

Reply:
This is to support clients (JS) that prefer to handle JSON than the binary representation. For golang, we actually cannot handle the JSON using the current requester because it will simply unmarshal the json into the struct and we can't unmarshal JSON directly into the block type due to the `Action` and `Auth` interfaces.

Have a note here https://github.com/ava-labs/hypersdk/pull/1606#issuecomment-2389723597 to add an encoding parameter that we could block this PR on if you prefer.